src/gemm.jl

```diff
@@ -147,17 +147,24 @@ const XTranspose{T} = Transpose{T, <:AbstractVecOrMat{T}}
 for TA in [:AbstractMatrix, :XTranspose]
     for TB in [:AbstractMatrix, :XTranspose]
         @eval function LinearAlgebra.mul!(o::AbstractMatrix{T}, a::$TA{T}, b::$TB{T}, α::Number, β::Number) where {T<:Tropical{<:NativeTypes}}
-            α = _convert_to_tropical(T, α)
-            β = _convert_to_tropical(T, β)
-            if iszero(β)
-                @avx for j=1:size(o, 2), i=1:size(o, 1)
-                    o[i,j] = zero(T)
-                end
-            end
+            α = _convert_to_static(T, α)
+            β = _convert_to_static(T, β)
             Octavian.matmul!(o, a, b, α, β)
         end
     end
 end
+# NOTE: benchmark shows, the type instability here can be optimized by the compiler
+# so you do not need to worry about the overheads.
```

Comment:
It still makes me nervous, but it is good at optimizing small unions. Most obviously, `for` loops all have unions.

But is this really what you want?
That alpha isn't allowed to be anything other than `0` or `1`?

Reply:
I compare this version with the explicit version with `if ... else ...`. Both cases does not introduce new allocations. So I think it is safe to do so.

Yes, I am sure this is what I want. A Tropical number is a semiring that can not be converted to/from a regular number. Here, I just want to to treat the zero element and one element properly. There is a correspondence of zero-element and one element in different algebras, but not for other numbers.